File: app/src/main/java/com/sakuna63/tumblrin/data/datasource/PostDataSource.java
package com.sakuna63.tumblrin.data.datasource;

import android.support.annotation.IntRange;

import com.sakuna63.tumblrin.data.model.Post;

import java.util.List;

import rx.Single;

public interface PostDataSource {
    Single<List<Post>> getDashboard(@IntRange(from = 1, to = 20) Integer limit, @IntRange(from = 0) Integer offset, Long sinceId);
}

File: app/src/main/java/com/sakuna63/tumbin/data/datasource/PostDataSource.kt
package com.sakuna63.tumbin.data.datasource

import android.support.annotation.IntRange

import rx.Completable
import rx.Single

interface PostDataSource {
    fun fetchDashboard(@IntRange(from = 1, to = 20) limit: Int?,
                       @IntRange(from = 0) offset: Int?,
                       sinceId: Long?,
                       beforeId: Long?): Single<Int>

    fun deleteAllDashboardCache(): Completable
}
